Hryhoriichuk.University.Instagram.Web/Views/Post/PostDetail.cshtml

```diff
@@ -0,0 +1,436 @@
+﻿@model Hryhoriichuk.University.Instagram.Web.Models.PostInfo
+
+@{
+    Layout = "_Layout";
+    var userManager = Model.UserManager;
+}
+
+<style>
+    /* Add your custom CSS styles here */
+    .modal {
+        display: none;
+        position: fixed;
+        z-index: 9999;
+        left: 0;
+        top: 0;
+        width: 100%;
+        height: 100%;
+        overflow: auto;
+        background-color: rgba(0,0,0,0.7);
+    }
+
+    .modal-content {
+        background-color: #fefefe;
+        margin: 10% auto;
+        padding: 20px;
+        border: 1px solid #888;
+        width: 80%;
+    }
+
+    .close {
+        color: #aaa;
+        float: right;
+        font-size: 28px;
+        font-weight: bold;
+    }
+
+        .close:hover,
+        .close:focus {
+            color: black;
+            text-decoration: none;
+            cursor: pointer;
+        }
+
+    .post-container {
+        border: 1px solid #ddd;
+        border-radius: 5px;
+        margin-bottom: 20px;
+        padding: 10px;
+    }
+
+    .post-header {
+        display: flex;
+        align-items: center;
+        margin-bottom: 10px;
+    }
+
+    .post-username {
+        margin-left: 10px;
+        font-weight: bold;
+        color: inherit;
+        text-decoration: none;
+    }
+
+    .post-profile-picture {
+        width: 30px;
+        height: 30px;
+        border-radius: 50%;
+        object-fit: cover;
+    }
+
+    .post-image {
+        max-width: 100%;
+        border-radius: 5px;
+        width: 1080px;
+    }
+
+    .post-caption {
+        margin-bottom: 10px;
+    }
+
+    .comment-container {
+        margin-top: 20px;
+    }
+
+    .comment {
+        margin-bottom: 10px;
+    }
+
+    .comment-username {
+        font-weight: bold;
+    }
+
+    .comment-input {
+        margin-top: 20px;
+    }
+
+    .dropdown-menu {
+        position: absolute;
+        right: 0;
+        background-color: #f9f9f9;
+        min-width: 160px;
+        box-shadow: 0 2px 5px rgba(0,0,0,0.1);
+        border-radius: 5px;
+        display: none;
+    }
+
+    .dropdown-item {
+        padding: 10px;
+        text-decoration: none;
+        display: block;
+        color: #333;
+    }
+
+        .dropdown-item:hover {
+            background-color: #ddd;
+        }
+
+    .show {
+        display: block;
+    }
+
+
+</style>
+
+
+@if (!(ViewData["IsCurrentUserProfile"] as bool? == true))
```

Comment:
Here you use a lot of enclosed conditions. Maybe it will be better to divide the logic into separate methods